src/mscorlib/src/System/Collections/Hashtable.cs

```diff
@@ -66,6 +66,7 @@ namespace System.Collections
     // 
     [DebuggerTypeProxy(typeof(System.Collections.Hashtable.HashtableDebugView))]
     [DebuggerDisplay("Count = {Count}")]
+    [Serializable]
```

Comment:
> More generally, are we actually exposing this Hashtable from any APIs, e.g. are we returning this from APIs that return IDictionary?

We have been engineering around this problem (e.g. Environment.GetVariables). I agree that we should finally bite the bullet and move Hashtable to CoreLib if there is any reason for it.